Modules/_abc.c

```diff
@@ -20,13 +20,27 @@ _Py_IDENTIFIER(_abc_impl);
 _Py_IDENTIFIER(__subclasscheck__);
 _Py_IDENTIFIER(__subclasshook__);
 
-/* A global counter that is incremented each time a class is
+typedef struct {
+    PyObject *_abc_data_type;
+} _abcmodule_state;
+
+/*
+   FIXME: PEP 573: Move abc_invalidation_counter into _abcmodule_state.
+   A global counter that is incremented each time a class is
    registered as a virtual subclass of anything.  It forces the
    negative cache to be cleared before its next use.
    Note: this counter is private. Use `abc.get_cache_token()` for
    external code. */
```

Comment:
nitpick: usually FIXME are written after the usual comment

```suggestion
/* A global counter that is incremented each time a class is
   registered as a virtual subclass of anything.  It forces the
   negative cache to be cleared before its next use.
   Note: this counter is private. Use `abc.get_cache_token()` for
   external code. */
// FIXME: PEP 573: Move abc_invalidation_counter into _abcmodule_state.
```